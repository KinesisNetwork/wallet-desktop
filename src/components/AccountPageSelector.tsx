import { AccountPage } from '@types'
import * as React from 'react'

export interface Props {
  accountPage: AccountPage
  setAccountPage: (act: AccountPage) => any
}

export const AccountPageSelector: React.SFC<Props> = ({ accountPage, setAccountPage }) => {
  const isActive = (view: AccountPage): string => accountPage === view ? 'is-active' : ''
  return (
    <div className='buttons'>
      <button
        className={`button ${isActive(AccountPage.dashboard)}`}
        onClick={() => setAccountPage(AccountPage.dashboard)}
      >
        <span className='icon'><i className='fas fa-home' /></span>
      </button>
      <button
        className={`button ${isActive(AccountPage.sign)}`}
        onClick={() => setAccountPage(AccountPage.sign)}
      >
        <span className='icon'><i className='fas fa-pencil-alt' /></span>
      </button>
    </div>
  )
}

AccountPageSelector.displayName = 'AccountPageSelector'
