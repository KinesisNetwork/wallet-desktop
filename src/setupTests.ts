import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
