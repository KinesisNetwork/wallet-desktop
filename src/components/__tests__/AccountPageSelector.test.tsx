import { shallow } from 'enzyme';
import * as React from 'react'

import { AccountPageSelector } from '@components/AccountPageSelector'
import { AccountPage as AccountPageEnum } from '@types'
import '../../setupTests'

describe('AccountPageSelector', () => {
  it('renders correctly', () => {
    const wrapperWhenDashboard = shallow(<AccountPageSelector accountPage={AccountPageEnum.dashboard} setAccountPage={() => null} />)
    const wrapperWhenSign = shallow(<AccountPageSelector accountPage={AccountPageEnum.sign} setAccountPage={() => null} />)

    expect(wrapperWhenDashboard).toMatchSnapshot()
    expect(wrapperWhenSign).toMatchSnapshot()
  })

  it('should render both transfer and sign buttons', () => {
    const wrapper = shallow(<AccountPageSelector accountPage={AccountPageEnum.dashboard} setAccountPage={() => null} />)

    const buttons = wrapper.find('button')

    expect(buttons).toHaveLength(2)
    expect(buttons.at(0).find('i').hasClass('fa-home')).toBe(true)
    expect(buttons.at(1).find('i').hasClass('fa-pencil-alt')).toBe(true)
  })

  it('makes dashboard button active', () => {
    const setAccountPageStub = jest.fn()
    const wrapper = shallow(<AccountPageSelector accountPage={AccountPageEnum.dashboard} setAccountPage={setAccountPageStub} />)

    const buttons = wrapper.find('button')
    const dashboardButton = buttons.at(0)
    const signButton = buttons.at(1)

    expect(dashboardButton.hasClass('is-active')).toBe(true)
    expect(signButton.hasClass('is-active')).toBe(false)

    dashboardButton.simulate('click')
    expect(setAccountPageStub).toHaveBeenCalledTimes(1)
    expect(setAccountPageStub).toHaveBeenCalledWith(AccountPageEnum.dashboard)
  })

  it('makes sign button active', () => {
    const setAccountPageStub = jest.fn()
    const wrapper = shallow(<AccountPageSelector accountPage={AccountPageEnum.sign} setAccountPage={setAccountPageStub} />)

    const buttons = wrapper.find('button')
    const dashboardButton = buttons.at(0)
    const signButton = buttons.at(1)

    expect(buttons).toHaveLength(2)
    expect(dashboardButton.hasClass('is-active')).toBe(false)
    expect(signButton.hasClass('is-active')).toBe(true)

    signButton.simulate('click')
    expect(setAccountPageStub).toHaveBeenCalledTimes(1)
    expect(setAccountPageStub).toHaveBeenCalledWith(AccountPageEnum.sign)
  })
})
