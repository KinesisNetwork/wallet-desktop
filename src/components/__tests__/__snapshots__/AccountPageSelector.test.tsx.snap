// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountPageSelector renders correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-active"
    onClick={[Function]}
  >
    <span
      className="icon"
    >
      <i
        className="fas fa-home"
      />
    </span>
  </button>
  <button
    className="button "
    onClick={[Function]}
  >
    <span
      className="icon"
    >
      <i
        className="fas fa-pencil-alt"
      />
    </span>
  </button>
</div>
`;

exports[`AccountPageSelector renders correctly 2`] = `
<div
  className="buttons"
>
  <button
    className="button "
    onClick={[Function]}
  >
    <span
      className="icon"
    >
      <i
        className="fas fa-home"
      />
    </span>
  </button>
  <button
    className="button is-active"
    onClick={[Function]}
  >
    <span
      className="icon"
    >
      <i
        className="fas fa-pencil-alt"
      />
    </span>
  </button>
</div>
`;
